Subject: [PATCH 1/3] Fix conftest to ignore implicit-function-declaration and
 strict-prototypes warnings

conftest rely on the fact that a missing prototype should build, but an invalid
call to a function (missing function parameters) the build should fail.
---
 conftest.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/conftest.sh b/conftest.sh
index ea2676e..6e6da83 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -101,7 +101,8 @@ test_header_presence() {
 build_cflags() {
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM"
+-nostdinc -isystem $ISYSTEM \
+-Wno-implicit-function-declaration -Wno-strict-prototypes"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
-- 
2.45.0

Subject: [PATCH 2/3] Fix conftest to use a short wchar_t

Fix build error about ``const efi_char16_t *v = L"SecureBoot"``
when including include/linux/efi.h
---
 conftest.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conftest.sh b/conftest.sh
index 6e6da83..678b79c 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -101,7 +101,7 @@ test_header_presence() {
 build_cflags() {
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
--nostdinc -isystem $ISYSTEM \
+-nostdinc -isystem $ISYSTEM -fshort-wchar \
 -Wno-implicit-function-declaration -Wno-strict-prototypes"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
-- 
2.45.0

Subject: [PATCH 3/3] Fix conftest to use nv_drm_gem_vmap() which has the
 secondary map argument
---
 conftest.sh | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/conftest.sh b/conftest.sh
index 678b79c..606f13d 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -4593,8 +4593,13 @@ compile_test() {
             #
             CODE="
             #include <drm/drm_gem.h>
+            #if defined(NV_LINUX_IOSYS_MAP_H_PRESENT)
+            typedef struct iosys_map nv_sysio_map_t;
+            #else
+            typedef struct dma_buf_map nv_sysio_map_t;
+            #endif
             int conftest_drm_gem_object_vmap_has_map_arg(
-                    struct drm_gem_object *obj, struct dma_buf_map *map) {
+                    struct drm_gem_object *obj, nv_sysio_map_t *map) {
                 return obj->funcs->vmap(obj, map);
             }"
 
-- 
2.45.0
